tools/rpkg/src/types.cpp

```diff
@@ -9,15 +9,15 @@
 using namespace duckdb;
 
 RType RApiTypes::DetectRType(SEXP v, bool integer64) {
-	if (TYPEOF(v) == REALSXP && Rf_inherits(v, "POSIXct")) {
+	if (IS_NUMERIC(v) && Rf_inherits(v, "POSIXct")) {
 		return RType::TIMESTAMP;
-	} else if (TYPEOF(v) == REALSXP && Rf_inherits(v, "Date")) {
+	} else if (IS_NUMERIC(v) && Rf_inherits(v, "Date")) {
 		return RType::DATE;
-	} else if (TYPEOF(v) == INTSXP && Rf_inherits(v, "Date")) {
+	} else if (IS_INTEGER(v) && Rf_inherits(v, "Date")) {
```

Comment:
Shouldn't integral tests come before numeric tests? Or are integers not numeric?

Reply:
They are not